src/assert.c

```diff
@@ -25,10 +25,10 @@ void vec_check_size(r_obj* x,
                     r_ssize size,
                     struct vctrs_arg* arg,
                     struct r_lazy call) {
-  r_ssize x_size = vec_size(x);
+  r_ssize x_size = vec_size_params(x, arg, call);
 
   if (x_size != size) {
-    stop_assert_size(x_size, size, arg);
+    stop_assert_size(x_size, size, arg, call);
```

Comment:
The R side of `stop_assert_size()` has a `call` argument, but we weren't taking advantage of it.

I'm not sure `vec_check_size()` was actually being used anywhere, which is part of the reason we haven't seen this